cirq-core/cirq/circuits/circuit_operation.py

```diff
@@ -259,7 +315,7 @@ def _measurement_key_objs_(self) -> AbstractSet['cirq.MeasurementKey']:
                 self,
                 '_cached_measurement_key_objs',
                 {
-                    protocols.with_measurement_key_mapping(key, self.measurement_key_map)
+                    protocols.with_measurement_key_mapping(key, dict(self.measurement_key_map))
```

Comment:
Let's file an issue to fix the `with_measurement_key_mapping` protocol to we can avoid the copy here and on line 361 below.

Reply:
Done: #5552. Unclear if this should be before or after 1.0 (may be more effort than it's worth likely breaking but not 100% certain), so I marked as triage-discuss.